src/nbsap/forms.py

```diff
@@ -216,24 +222,49 @@ def __init__(self, *args, **kwargs):
         self.fields['title'].initial = title
         self.fields['description'].initial = description
         self.fields['language'].initial = lang
+        if self.activity:
+            self.fields['code'].initial = self.activity.code
 
     def save(self):
         activity = self.activity or EuAction()
         lang = self.cleaned_data['language']
         title = self.cleaned_data['title']
+        code = self.cleaned_data.get('code', None)
         description = self.cleaned_data['description']
         setattr(activity, 'title_%s' % lang, title)
         setattr(activity, 'description_%s' % lang, description)
         activity.parent = self.parent
-        activity.code = activity.code or activity.get_next_code()
+        activity.code = code or activity.get_next_code(self.target)
 
         activity.save()
         if not self.parent:
             activity.target = [self.target]
             activity.save()
+
+        if activity.subactions():
```

Comment:
Don't check. activity.subactions will always return an iterable, even though it is [] the empty list.